### PR TITLE
#120 - fix load() in DataImporter.php

### LINKED ISSUE #120: Fix load() in DataImporter.php
When city exists in database, method doesn't check if it's in the right country.
It creates problem, cause there are cities that are named the same, but located in different countries.

### PR DESCRIPTION
**Pull request by AleksandraKozubal**
Modified load(), it checks if city is assigned to given country.
Also there can be multiple cities that have the same name, so changed migration.

It should close #120

### CODE REVIEW

File: app/Importers/BinBinDataImporter.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+class BinBinDataImporter extends DataImporter
+{
+    protected Crawler $sections;
+
+    public function extract(): static
+    {
+        try {
+            $response = $this->client->get("https://www.binbin.tech/lokasyonlar");
+            $html = $response->getBody()->getContents();
+        } catch (GuzzleException) {
```
Comment (kamilpiech97) [inline app/Importers/BinBinDataImporter.php:19]: What do we do with the caught exception?
Maybe we would store that exception as a meta information in import info details or smth like that.

Comment (AleksandraKozubal) [reply app/Importers/BinBinDataImporter.php]: this commit was placed here by accident, actual BinBin is in #117

File: app/Importers/DataImporter.php [hunk 2]
```diff
@@ -12,16 +12,22 @@
 use App\Models\ImportInfoDetail;
 use App\Services\MapboxGeocodingService;
 use GuzzleHttp\Client;
+use Stichoza\GoogleTranslate\GoogleTranslate;
 
 abstract class DataImporter
 {
+    public const language = "en";
```
Comment (mtracz) [inline app/Importers/DataImporter.php:19]: Const should be UPPERCASE.
(StringManipulation is a good plugin for PHPStorm).

Do you need this constant as `public`? Is it used anywhere else?

File: app/Importers/DataImporter.php [hunk 3]
```diff
@@ -12,16 +12,22 @@
...
     protected bool $stopExecution = false;
     protected int $importInfoId;
+    protected GoogleTranslate $translate;
 
     public function __construct(
         protected Client $client,
         protected MapboxGeocodingService $mapboxService,
-    ) {}
+    ) {
+        $this->translate = new GoogleTranslate();
```
Comment (mtracz) [inline app/Importers/DataImporter.php:29]: Why not in the constructor?

File: app/Importers/DataImporter.php [hunk 4]
```diff
@@ -30,6 +36,11 @@ public function setImportInfo(int $importInfoId): static
         return $this;
     }
 
+    public function translate(string $word, $language): string
```
Comment (mtracz) [inline app/Importers/DataImporter.php:39]: Missed type for `$language`.

File: app/Importers/DataImporter.php [hunk 5]
```diff
@@ -99,19 +110,26 @@ protected function createImportInfoDetails(string $code, string $providerName):
...
-        if ($city || $alternativeCityName) {
-            $cityId = $city ? $city->id : $alternativeCityName->city_id;
+        if ($country) {
+            $city = City::query()->where("name", $cityName)->where("country_id", $country->id)->first();
+            $alternativeCityName = CityAlternativeName::query()->where("name", $cityName)->first();
 
-            $this->createProvider($cityId, self::getProviderName());
+            if ($city) {
+                $cityId = $city->id;
+                $this->createProvider($cityId, self::getProviderName());
```
Comment (mtracz) [inline app/Importers/DataImporter.php:121]: Why method `getProviderName()` is static?

**Review by mtracz — CHANGES_REQUESTED** [review]

File: app/Importers/DataImporter.php (hunk 2 above)
Comment (AleksandraKozubal) [reply app/Importers/DataImporter.php:19]: Yeah, i should make it private, but thanks to this comment I noticed that it should be deleted - DataImporter is not usinh GooggleTranslate anymore

File: app/Importers/DataImporter.php (hunk 3 above)
Comment (AleksandraKozubal) [reply app/Importers/DataImporter.php:29]: same as above

File: app/Importers/DataImporter.php (hunk 4 above)
Comment (AleksandraKozubal) [reply app/Importers/DataImporter.php:39]: same as above

File: app/Importers/DataImporter.php (hunk 3 above)
Comment (AleksandraKozubal) [reply app/Importers/DataImporter.php]: and the reason to having it here was this problem with tests - i gave 3 arguments, but it said that i gave only 2
https://github.com/blumilksoftware/escooters/actions/runs/5887951907/job/15968210359

File: app/Importers/DataImporter.php (hunk 5 above)
Comment (AleksandraKozubal) [reply app/Importers/DataImporter.php]: Changed it
I wanted to use `get_class($this)` but composer forces `static::class`

Comment (mtracz) [reply app/Importers/DataImporter.php]: Ok fine. This change will break all importers.

Comment (AleksandraKozubal) [reply app/Importers/DataImporter.php:110]: Undone it

Comment (mtracz) [reply app/Importers/DataImporter.php:110]: Lets keep it as it is.

**Review by mtracz — APPROVED** [review]

**Review by kamilpiech97 — APPROVED** [review]

**Review by Letha0 — APPROVED** [review]

**Pull request merged**